pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -1304,15 +1314,30 @@ func (n *NetworkPolicyController) syncAppliedToGroup(key string) error {
 
 // getAppliedToWorkloads returns a list of workloads (Pods and ExternalEntities) selected by an AppliedToGroup
 // for standalone selectors or corresponding to a ClusterGroup.
-func (n *NetworkPolicyController) getAppliedToWorkloads(g *antreatypes.AppliedToGroup) ([]*v1.Pod, []*v1alpha2.ExternalEntity) {
-	// Check if an internal Group object exists corresponding to this AppliedToGroup.
+func (n *NetworkPolicyController) getAppliedToWorkloads(g *antreatypes.AppliedToGroup) ([]*v1.Pod, []*v1alpha2.ExternalEntity, error) {
+	// Check if an internal Group object exists corresponding to this AppliedToGroup
 	group, found, _ := n.internalGroupStore.Get(g.Name)
 	if found {
 		// This AppliedToGroup is derived from a ClusterGroup.
 		grp := group.(*antreatypes.Group)
-		return n.getInternalGroupWorkloads(grp)
+		pods, ees := n.getInternalGroupWorkloads(grp)
```

Comment:
We could potentially add an optional namespace check in `GetEntities` used by `getInternalGroupWorkloads`, so that we can return early and don't need to loop thru all entities again after getting a list of them. Doesn't need to be address in this PR; a TODO will suffice

Reply:
Thanks for the suggestion. I moved the namespace check into `getInternalGroupWorkloads`. I prefer not to add custom checks to `GetEntities`, which is a generic interface while checking namespace is a too business specific condition. There are other similar business specific checking/filtering and currently all of them are performed on caller side.